packages/client/hmi-client/tests/unit/utils/number.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { describe, expect, it } from 'vitest';
+import { exponentialToNumber, numberToNist, nistToNumber, displayNumber } from '@/utils/number';
+
+describe('number util tests', () => {
+	describe('exponentialToNumber', () => {
+		it('should correctly convert very large positive exponents', () => {
+			expect(exponentialToNumber('2e+100')).to.equal(`2${'0'.repeat(100)}`);
+			expect(exponentialToNumber('3.45e+100')).to.equal(`345${'0'.repeat(98)}`);
+		});
+
+		it('should correctly convert very large negative exponents', () => {
+			expect(exponentialToNumber('-2e+100')).to.equal(`-2${'0'.repeat(100)}`);
+			expect(exponentialToNumber('-3.45e+100')).to.equal(`-345${'0'.repeat(98)}`);
+		});
+
+		it('should return the original string for non-exponential numbers', () => {
+			expect(exponentialToNumber('456')).to.equal('456');
+			expect(exponentialToNumber('4.56')).to.equal('4.56');
+		});
+
+		it('should return "NaN" for non-numeric strings', () => {
+			expect(exponentialToNumber('def')).to.equal('NaN');
+			expect(exponentialToNumber('4.56def')).to.equal('4.56');
+		});
+
+		// Additional test cases
+		it('should correctly convert zero exponent', () => {
+			expect(exponentialToNumber('2e0')).to.equal('2');
+			expect(exponentialToNumber('-2e0')).to.equal('-2');
+		});
+
+		it('should correctly convert one exponent', () => {
+			expect(exponentialToNumber('2e1')).to.equal('20');
+			expect(exponentialToNumber('-2e1')).to.equal('-20');
+		});
+	});
+
+	describe('numberToNist', () => {
```

Comment:
Need example here and `nistToNumber` and `displayNumber` for this rule
![image](https://github.com/DARPA-ASKEM/terarium/assets/1030720/f123350a-ea6a-48c1-9215-d2eba0a76cb5)

>However, when there are only four digits before or after the decimal marker, we recommend that no space is required and no space should be used

For example:
`1234.123` -> `1234.123`
`1234.1234` -> `1234.1234`
`11234.123` -> `11 234.123`
`1234.12344` -> `1 234.123 44`
`11234.123` -> `11 234.123`
`11234.1234` -> `11 234.123 4`